### ISSUE TITLE
Recommended strategy for stubbing entities in unit tests in version 27.0.0

### ISSUE THREAD

**Issue by samcic** [stripe/stripe-dotnet#1651]
OS version: Windows 10
.NET Core version: 2.2
Stripe.net version: 27.0.0

Before version 27.0.0 it was trivial to create unit tests with mocking/stubbing behavior, e.g.

```
[TestMethod]
public async Task Example()
{
    Moq<ISomeServiceThatMakesStripeApiRequest>().Setup(m => m.GetCustomer("cus_x")).ReturnsAsync(new Customer{DefaultSourceId = "src_x"});
    var result = await _target.SomeWrapperMethodAsync("cus_x");
    ...
}
```

In version 27.0.0 the method above no longer compiles because many of the core resources (e.g. Customer) now contain properties that are readonly.

What's the recommended approach for creating stubbed resources (e.g. a `Customer` with `DefaultSourceId` set to a test value) now? Of course we could create our own wrapper objects for all your entities, but this would be a maintenance pain.

We're concerned that you guys have neglected unit-test considerations with version 27.0.0.

**Comment by TheNowyn** [comment]
Fully agree.

**Comment by ob-stripe** [comment]
Hi, thanks for the report and sorry for the inconvenience. We changed the way expandable fields are handled, and in the process removed setters for the ID and expanded versions of these fields, but failed to realize these setters were useful for unit tests/mocking. We'll add the setters back and publish a new version soon.

**Comment by ob-stripe** [comment]
Fixed in 27.1.0.

**Comment by TheNowyn** [comment]
While setters are back and code compiles, when trying to set a Customer or CustomerId properties of say a Charge throws NullReferenceException.

Are charges being tested at all before releasing?
